admin/app/components/solidus_admin/ui/table/component.rb

```diff
@@ -1,10 +1,11 @@
 # frozen_string_literal: true
 
 class SolidusAdmin::UI::Table::Component < SolidusAdmin::BaseComponent
-  def initialize(rows:, model_class:, columns: [])
+  def initialize(rows:, model_class:, columns: [], pagination_component: nil)
```

Comment:
Coupling the table component with a page at this point is probably worth it. We have the `rows` and the `pagination_component` parameters requiring the caller to extract information from a page. I also wonder if extracting the `model_class` also from a page is possible (maybe not if the collection is empty?). Do you think we can explore that solution? It also would make the code more consistent, as we're providing component classes in the initializer but not component instances as in this case.

If we go the extra mile, we could probably split the component into two: the one coupled with a `page`, and a pure UI one. But we can wait for this one until we find its need.

Reply:
Thank you for bringing up this point @waiting-for-dev 🙏 

>Coupling the table component with a page at this point is probably worth it. We have the rows and the pagination_component parameters requiring the caller to extract information from a page.
Do you think we can explore that solution? It also would make the code more consistent, as we're providing component classes in the initializer but not component instances as in this case.

>If we go the extra mile, we could probably split the component into two: the one coupled with a page, and a pure UI one. But we can wait for this one until we find its need.

So you're thinking about having something like this at the end, right?
```ruby
def initialize(page:, columns: [], pagination_component: component("ui/pagination"))
  .
  .
  .
end
  ```

I think in general it could be a good idea for now 🤔
We can start by coupling these logics and exploring the possibilities of a more generic table component in the future.

My main concern is how to determine the path (e.g., `product_path` in this example) that I'm currently passing as a proc to the pagination component.

We can consider the following solution:
1. Create a separate method in the table component to handle generating the path based on the provided proc.
2. Use a specific approach to generate the path. For example, if the table component is used for displaying `Spree::Product`s, we can assume that the path helper would follow the pattern `product_path`. This way, we can automatically generate the path without explicitly passing it as a proc.
3. Allow the flexibility to override the path generation logic by accepting a lambda as an optional argument to the table component. This way, users can provide their own path generation logic if needed.

The last point raises my main concern that by simplifying the configuration too much, we risk creating a less reusable table component since it is a UI component.


> I also wonder if extracting the model_class also from a page is possible (maybe not if the collection is empty?).

Yeah, I have to explore this point better, but I think it's possible 👍 

Let me know WDYT 🙌

Reply:
> So you're thinking about having something like this at the end, right?
> 
> ```ruby
> def initialize(page:, columns: [], pagination_component: component("ui/pagination"))
>   .
>   .
>   .
> end
> ```

Yeah! That would be great :slightly_smiling_face:

> * Create a separate method in the table component to handle generating the path based on the provided proc.
> * Use a specific approach to generate the path. For example, if the table component is used for displaying `Spree::Product`s, we can assume that the path helper would follow the pattern `product_path`. This way, we can automatically generate the path without explicitly passing it as a proc.
> * Allow the flexibility to override the path generation logic by accepting a lambda as an optional argument to the table component. This way, users can provide their own path generation logic if needed.

I think I'd go with the third option, dispatching the argument from the table to the pagination component. As you said, we could add some conventions to make the discovery automatic, but I won't be bothered if we need to pass the proc every time :+1:

> The last point raises my main concern: by simplifying the configuration too much, we risk creating a less reusable table component since it is a UI component.

I completely agree. We're now making explicit the coupling we already had (having three parameters that were deducible from a page). A nice next step would be creating a pure table UI component, without knowledge of a page or ActiveRecord. Then the current one should compose the UI one.

Reply:
> A nice next step would be creating a pure table UI component, without knowledge of a page or ActiveRecord. Then the current one should compose the UI one.

Definitely, that will be the next step. I don't want to continue these changes in this PR to avoid deviating too much from the integration purpose. However, once we have the need, we should definitely proceed with something like that.